### PR TITLE
Refactor very, very many things (primarily to reduce duplication)

### PR DESCRIPTION
**Pull request by ricky-wong**
- (Primary) Remove location-specific duplicated code everywhere
- Introduce `timeDifferential` and `timeString` as utility functions
- Construct objects with a single object parameter, to make fields clear
- Favor building strings functionally instead of imperatively
- Prefer `const` and `let`
- Comment TODOs and weird gotchas
- Reverse arguments to `pingChannel` to have the shorter argument first
- Preliminary support for different time zones (clients and arcades)
- Rename environment variables to `DISCORD_BOT_TOKEN` and `EACOOKIE_shopid`
- Remove unused `Player.location` and `getRinonMessage()`
- Don't attempt `updateChannelTopics()` until we've logged in
- `!whose` scolds you now. `!here` gives you the old behavior (still discouraged in favor of checking the channel topic)

### CODE REVIEW

File: bot.js [hunk 1]
```diff
@@ -7,504 +7,440 @@ require('dotenv').config();
...
 const msMinute = 60*1000;
 const msHour = 60*60*1000;
+const REFRESH_INTERVAL = msMinute;
+function timeDifferential(nowTime, beforeTime) {
+  const hr = Math.floor((nowTime - beforeTime) / msHour);
+  const min = Math.floor(((nowTime - beforeTime) % msHour) / msMinute);
+  return {
+    h: hr,
+    m: min,
+    str: `${hr}h${min}m`
```
Comment (korisya) [inline bot.js]: I prefer a space between the hours and minutes.
```javascript
str: `${hr}h ${min}m`
```

File: bot.js [hunk 2]
```diff
@@ -7,504 +7,440 @@ require('dotenv').config();
...
-			channel.send("```" + milpitas.getAll() + "```");
-		} else if (locName === 'Daly City' && channel.name === 'dnb-dalycity') {
-			channel.send("```" + dalyCity.getAll() + "```");
-		} else if (locName === 'Concord' && channel.name === 'round1-concord') {
-			channel.send("```" + concord.getAll() + "```");
-		} else if (locName === 'San Jose' && channel.name === 'round1-sanjose') {
-			channel.send("```" + sanJose.getAll() + "```");
-		}
-	});
+function yeet(loc) {
```
Comment (korisya) [inline bot.js]: rename this lol

File: bot.js [hunk 3]
```diff
@@ -7,504 +7,440 @@ require('dotenv').config();
...
+    }
+  });
+
+  let topic;
+  if (loc.todaysPlayers.length === 0) {
+    // TODO: Perhaps include the day's start time (local time per arcade)
+    topic = `${nowString}: 0 players today.`;
+  } else if (numPlayers === 0) {
+    const timeSinceSeen = timeDifferential(currentTime, loc.todaysPlayers[0].lastTime);
+    topic = `${nowString}: Everyone(${loc.todaysPlayers.length}) left! :eyes: Last player seen: ${loc.todaysPlayers[0].name} ${timeSinceSeen.str} ago.`;
```
Comment (korisya) [inline bot.js]: Prefer space between `Everyone` and the total number of today's players.
```javascript
topic = `${nowString}: Everyone (${loc.todaysPlayers.length}) left!...`;
```

Comment (korisya) [reply bot.js]: Also kind of strange to put it after `Everyone`. It's not immediately clear what the number means.

Comment (ricky-wong) [reply bot.js]: It was ambiguous on purpose, I just wanted to drop it in to see if it was useful. You're right, though. How about something like "`All 8 players left!`"?

Comment (korisya) [reply bot.js]: It kind of sounds like there were 8 players together at one point... maybe `All 8 players today left!`

**Pull request merged**